### ISSUE TITLE
New-MgTeam cmdlet does not work

### ISSUE THREAD

**Issue by bchap1n** [microsoftgraph/msgraph-sdk-powershell#119]
I am able to create new Teams using the Graph Explorer:

`POST:  https://graph.microsoft.com/beta/Teams
{"template@odata.bind":"https://graph.microsoft.com/beta/teamsTemplates('standard')","displayName":"waffleSloth4","description":"waffleSloth4"}`

but when I try the same with 'New-mgTeam' it fails:

```
Connect-Graph -Scopes "Group.ReadWrite.All"

$body = @"
{
    "template@odata.bind":"https://graph.microsoft.com/beta/teamsTemplates('standard')",
    "displayName":"waffleSloth4",
    "description":"waffleSloth4"
}
"@

New-mgTeam -BodyParameter $body
```
error message:
`The navigation bind for template was missing in request.`

**Comment by peombwa** [comment]
@bchap1n As part of #87,  we will be introducing an `AdditionalProperties` bag that will allow customers to pass extra properties to the request body.

In this case, the request will look like this:
``` ps
$additionalProperties = @{
    "template@odata.bind" = "https://graph.microsoft.com/beta/teamsTemplates('standard')"
}
$body = @{
    "additionalProperties": $additionalProperties,
    "displayName":"waffleSloth4",
    "description":"waffleSloth4"
}

New-mgTeam -BodyParameter $body
```

**Comment by darrelmiller** [comment]
@peombwa  I think we need to add this "additionalProperties" to every entity, regardless of whether OpenType is true or not.  Also, in the other SDKs we call it "additionalData" not "additionalProperties".  Let's discuss further.

**Comment by KirkMunro** [comment]
My recommendation would be as follows:

1. Add a common `-PropertyValues` parameter of type `IDictionary`.
1. Make the use of that parameter possible without having to muck around with JSON or the `-BodyParameter` (i.e. leave `-BodyParameter` for the most advanced cases where the cmdlet parameters don't work for you, and use the explicitly defined parameters (e.g. `-DisplayName`, `-Description`) combined with the flexible `-PropertyValues` parameter (which is the property bag) to build the request body that will be sent to Graph.
1. Rename `-BodyParameter` as `-RequestBody`, and give it two aliases: `BodyParameter` and `Body`. `RequestBody` is a more accurate and descriptive name, and `-Body` is a parameter on `Invoke-WebRequest` and `Invoke-RestMethod`, so there is familiarity and ease of discovery. The "Parameter" portion of `-BodyParameter` seems irrelevant given that it is a PowerShell parameter.
1. Eliminate the need for unnecessarily complex syntax when associating properties with existing entities. i.e. Instead of `"template@odata.bind" = "https://graph.microsoft.com/beta/teamsTemplates('standard')"`, which is a JSON way of indicating that you want to apply a built-in template to a team, simply use `-Template $template` or `-PropertyValues @{Template = $template}`, and handle the conversion to JSON internally.

With this in place, assuming I already did something like this:

```powershell
# Note: This does not work at the moment
$standardTemplate = Get-MgTeam -Name Standard
```

I could then do this:

```powershell

New-MgTeam -DisplayName waffleSloth4 -Description waffleSloth4 -Template $standardTemplate
```

or this:

```powershell
New-MgTeam -DisplayName waffleSloth4 -Description waffleSloth4 -PropertyValues @{
    Template = $standardTemplate
}
```

or this:

```powershell
New-MgTeam -PropertyValues @{
    DisplayName = 'waffleSloth4'
    Description = 'waffleSloth4'
    Template = $standardTemplate
}
```

or if I want access to the raw request, I could do this:

```powershell
$body = @"
{
    "template@odata.bind":"https://graph.microsoft.com/beta/teamsTemplates('standard')",
    "displayName":"waffleSloth4",
    "description":"waffleSloth4"
}
"@
New-MgTeam -RequestBody $body
```

These would all work. The first three would simply build the JSON request body that you are sending to Graph to do the work in the last one, rather than have you craft it yourself.

Also, the idea of "additionalProperties" or "additionalData" seems unnecessary here, especially given that the Graph endpoint doesn't need or use the extra nesting of properties when it is invoked like this:

```none
POST: https://graph.microsoft.com/beta/Teams {"template@odata.bind":"https://graph.microsoft.com/beta/teamsTemplates('standard')","displayName":"waffleSloth4","description":"waffleSloth4"}
```

Why add an extra level of nesting when Graph itself doesn't need it?

**Comment by RyanMerton** [comment]
Could you just fix the -Template parameter rather than adding a new feature AdditionalProperties? The property is already explicit. If you do however require Template in AdditionalProperties, please remove the -Template parameter.

**Comment by peombwa** [comment]
This has be fixed in v1.0,  `Install-Module Microsoft.Graph -AllowClobber -Force`.

``` ps
New-MgTeam -DisplayName $TeamName -Description $TeamDescription `
    -AdditionalProperties @{
        "template@odata.bind" = "https://graph.microsoft.com/beta/teamsTemplates('standard')"
    }
```

Updated teams samples can be found here - https://github.com/microsoftgraph/msgraph-sdk-powershell/blob/dev/samples/5-Teams.ps1.

**Comment by kpinel** [comment]
I know this is an old post but what about adding additional owners? You know, automation and stuff.
It just seems that we are adding graph structure rather than doing this using cmdlet parameters and having the module to the JSON transform
